v2/api/network/v1alpha1api20201101/virtual_network_types_gen.go

```diff
@@ -335,112 +340,165 @@ type VirtualNetwork_Status struct {
 	//the virtual network.
 	EnableVmProtection *bool `json:"enableVmProtection,omitempty"`
 
-	//Etag: A unique read-only string that changes whenever the resource is updated.
-	Etag *string `json:"etag,omitempty"`
-
 	//ExtendedLocation: The extended location of the virtual network.
-	ExtendedLocation *ExtendedLocation_Status `json:"extendedLocation,omitempty"`
-
-	//Id: Resource ID.
-	Id *string `json:"id,omitempty"`
+	ExtendedLocation *ExtendedLocation `json:"extendedLocation,omitempty"`
 
 	//IpAllocations: Array of IpAllocation which reference this VNET.
-	IpAllocations []SubResource_Status `json:"ipAllocations,omitempty"`
+	IpAllocations []SubResource `json:"ipAllocations,omitempty"`
 
 	//Location: Resource location.
 	Location *string `json:"location,omitempty"`
 
-	//Name: Resource name.
-	Name *string `json:"name,omitempty"`
-
-	//ProvisioningState: The provisioning state of the virtual network resource.
-	ProvisioningState *ProvisioningState_Status `json:"provisioningState,omitempty"`
+	// +kubebuilder:validation:Required
+	Owner genruntime.KnownResourceReference `group:"resources.azure.com" json:"owner" kind:"ResourceGroup"`
 
-	//ResourceGuid: The resourceGuid property of the Virtual Network resource.
-	ResourceGuid *string `json:"resourceGuid,omitempty"`
+	//Reference: Resource ID.
+	Reference *genruntime.ResourceReference `armReference:"Id" json:"reference,omitempty"`
 
 	//Subnets: A list of subnets in a Virtual Network.
-	Subnets []Subnet_Status_VirtualNetwork_SubResourceEmbedded `json:"subnets,omitempty"`
+	Subnets []genruntime.ResourceReference `json:"subnets,omitempty"`
```

Comment:
I _think_ we actually just want to remove this entirely. Might be worth syncing up with me on this though. My current thinking is that these shouldn't exist on the k8s type but should exist on the ARM type, and then we'll write an extension that populates them on the ARM type just before sending to Azure, to deal with #1944.

Although, doing this might solve the problem (by forcing the users to manage their vnets + subnets as one). I am actually not sure what ARM does with a payload that looks like this though, as presumably this ends up being just a collection of ARM IDs rather than a full subnet specification. It's possible that they then clear out other fields on the subnet because they were "unset" here?